sys/include/net/nanocoap.h

```diff
@@ -395,6 +395,22 @@ static inline unsigned coap_get_token_len(const coap_pkt_t *pkt)
     return (pkt->hdr->ver_t_tkl & 0xf);
 }
 
+/**
+ * @brief   Get pointer to a message's token
+ *
+ * @param[in]   pkt   CoAP packet
+ *
+ * @returns     pointer to the token, or NULL if there is no token
+ */
+static inline uint8_t *coap_get_token(const coap_pkt_t *pkt)
+{
+    if (coap_get_token_len(pkt)) {
+        return (uint8_t*)pkt->hdr + sizeof(coap_hdr_t);
+    } else {
+        return NULL;
+    }
```

Comment:
since any function that is dealing tokens need to know about its size there is no reason to have this function do that (the position of the zero length token is still right behind the hdr)
```suggestion
        return (uint8_t*)pkt->hdr + sizeof(coap_hdr_t);
```